doc/src/manual/interfaces.md

```diff
@@ -233,7 +233,7 @@ ourselves, we can officially define it as a subtype of an [`AbstractArray`](@ref
 | `similar(A, dims::Dims)`                        | `similar(A, eltype(A), dims)`          | Return a mutable array with the same element type and size *dims*                     |
 | `similar(A, ::Type{S}, dims::Dims)`             | `Array{S}(undef, dims)`                | Return a mutable array with the specified element type and size                       |
 | **Non-traditional indices**                     | **Default definition**                 | **Brief description**                                                                 |
-| `axes(A)`                                    | `map(OneTo, size(A))`                  | Return a tuple of `AbstractUnitRange{<:Integer}` of valid indices                    |
+| `axes(A)`                                    | `map(OneTo, size(A))`                  | Return a tuple of `AbstractUnitRange{<:Integer}` of valid indices. The axes should be their own axes, that is `axes.(axes(A),1) == axes(A)` should be satisfied. |
```

Comment:
```suggestion
| `axes(A)`                                    | `map(OneTo, size(A))`                  | Return a tuple of `AbstractUnitRange{<:Integer}` of valid indices. The axes should be their own axes, that is `only.(axes.(axes(A))) == axes(A)` should be satisfied. |
```
Maybe we could use `only` here? Otherwise, this would pass for axes that have multi-dimensional axes, as long as the first dimension matches.

Reply:
This would work, but I don't fully understand the use case: axes that have multidimensional axes? Are you suggesting that this is something that we should disallow with this requirement in the interface? Currently, the axes should be `AbstractUnitRange`s, so they should have only one axis.

Reply:
I don’t have a use case in mind and I don’t think we need to disallow axes with multidimensional axes explicitly. I just thought it would be better to make the condition more restrictive. Now that I see that axes are required to be `AbstractUnitRange`s, it doesn’t really make a difference. Feel free to ignore the suggested change.